crates/m3/src/gadgets/hash/keccak/mod.rs

```diff
@@ -173,6 +207,14 @@ impl Keccakf {
 							batch_2_state_out[TRACKS_PER_BATCH * k + track]
 						);
 					}
+					// Populating the packed and selected input and output columns.
+					let mut input: std::cell::RefMut<'_, [u64]> =
+						index.get_mut_as(self.input[(x, y)])?;
+					let mut output: std::cell::RefMut<'_, [u64]> =
+						index.get_mut_as(self.output[(x, y)])?;
+
+					input[k] = pts[k].per_batch(0)[0].state_in[(x, y)];
```

Comment:
`per_batch` is meant to help to clean the code at call site. It did help, IMO, as shown by `brt` usages below, however, here it doesn't and this looks ugly, IMO.

I'd suggest implementing two helpers to get the first state in and the last state out on the trace itself!